src/observables/OptionStream.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Options } from "../omnisharp/options";
+import { Subject } from "rxjs/Subject";
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { vscode } from "../vscodeAdapter";
+import 'rxjs/add/operator/take';
+import Disposable from "../Disposable";
+
+export default class OptionStream {
+    private optionStream: BehaviorSubject<Options>;
+    private optionChangeDebouncer: Subject<void>;
+
+    constructor(vscode: vscode) {
+        this.optionStream = new BehaviorSubject<Options>(Options.Read(vscode));
+        this.optionChangeDebouncer = new Subject<void>();
+        this.optionChangeDebouncer.debounceTime(1000).subscribe(() => { this.optionStream.next(Options.Read(vscode)); });
+    }
+
+    public post() {
+        this.optionChangeDebouncer.next();
+    }
+
+    public subscribe(observer: (options: Options) => void): Disposable {
+        return new Disposable(this.optionStream.subscribe(observer));
+    }
+
+    public async GetLatestOptions(): Promise<Options> {
```

Comment:
I would just call this `Options()`

Reply:
I don't think this needs to be promise or async. It looks like there's always synchronously a value in the BehaviorSubject?

Reply:
take(1) returns an `Observable(Options)` and we need to convert it to a Promise in order to be able to get the options.